test/SparqlAntlrParserTest.cpp

```diff
@@ -1027,21 +1013,10 @@ TEST(SparqlParser, propertyListPathNotEmpty) {
   expectPropertyListPathFails("<bar> [ <foo> ?bar ]");
 }
 
-namespace {
-template <typename Exception = ParseException>
-void expectTriplesSameSubjectPathFails(const string& input) {
-  EXPECT_THROW(parseTriplesSameSubjectPath(input), Exception) << input;
-}
-}  // namespace
-
 TEST(SparqlParser, triplesSameSubjectPath) {
-  auto expectTriples =
-      [](const string& input,
-         const std::vector<ad_utility::sparql_types::TripleWithPropertyPath>&
-             triples) {
-        expectCompleteParse(parseTriplesSameSubjectPath(input),
-                            testing::Eq(triples));
-      };
+  auto expectTriples = makeExpectCompleteParserEq<
+      parseTriplesSameSubjectPath,
```

Comment:
Yes I would ideally have the interface as
```
template <auto parserFunction, auto defaultMatcher = testing::Eq, typename... Values = ...>
makeExpectCompleteParser(PrefixMap ...
```
though I think that the infered default for the `Values` parameter pack will be difficult.
This would allow your code but also the same for `GraphPattern`: `makeExpectCompleteParser<parseGroupGraphPattern, IsGraphPattern, bool, vector<SparqlFilter>, auto>`.